packages/ra-core/src/controller/useListParams.ts

```diff
@@ -0,0 +1,282 @@
+import { useCallback, useEffect, useState } from 'react';
+// @ts-ignore
+import { useSelector, useDispatch } from 'react-redux';
+import { parse, stringify } from 'query-string';
+import { push } from 'connected-react-router';
+import lodashDebounce from 'lodash/debounce';
+import isEqual from 'lodash/isEqual';
+import pickBy from 'lodash/pickBy';
+import { Location } from 'history';
+
+import queryReducer, {
+    SET_FILTER,
+    SET_PAGE,
+    SET_PER_PAGE,
+    SET_SORT,
+    SORT_ASC,
+} from '../reducer/admin/resource/list/queryReducer';
+import { crudGetList } from '../actions/dataActions';
+import { changeListParams, ListParams } from '../actions/listActions';
+import { Sort, ReduxState, Identifier, RecordMap } from '../types';
+import removeEmpty from '../util/removeEmpty';
+import removeKey from '../util/removeKey';
+
+interface Props {
+    filter?: object;
+    filterDefaultValues?: object;
+    perPage?: number;
+    sort?: Sort;
+    location: Location;
+    resource: string;
+    debounce?: number;
+}
+
+interface Query extends ListParams {
+    data: RecordMap;
+    filterValues: object;
+    ids: Identifier[];
+    total: number;
+    displayedFilters: {
+        [key: string]: boolean;
+    };
+    requestSignature: any[];
+}
+
+interface Actions {
+    changeParams: (action: any) => void;
+    setPage: (page: number) => void;
+    setPerPage: (pageSize: number) => void;
+    setSort: (sort: Sort) => void;
+    setFilters: (filters: any) => void;
+    hideFilter: (filterName: string) => void;
+    showFilter: (filterName: string, defaultValue: any) => void;
+}
+
+const useListParams = ({
+    resource,
+    location,
+    filterDefaultValues,
+    sort = {
+        field: 'id',
+        order: SORT_ASC,
+    },
+    perPage = 10,
+    filter,
+    debounce = 500,
+}: Props): [Query, Actions] => {
+    const [displayedFilters, setDisplayedFilters] = useState({});
```

Comment:
This hoos should only contain the query-related logic, so I think that displayed filters, data from the provider, and the initial effect should stay in the controller (or go to another hook).

Reply:
About the filters, that would mean moving the filters actions out too

Reply:
right

Reply:
Maybe too soon to do that ? What about renaming them `activeFilters` and `setActiveFilters` ?